pkg/cli/start.go

```diff
@@ -1417,13 +1419,68 @@ func doShutdown(
 		return errors.Wrap(err, "error sending drain request")
 	}
 	for {
-		if _, err := stream.Recv(); err != nil {
+		resp, err := stream.Recv()
+		if err != nil {
 			if tolerateConnClosedByServer && grpcutil.IsClosedConnection(err) {
 				return nil
 			}
 			// Unexpected error; the caller should try again (and harder).
 			return errTryHardShutdown{err}
 		}
+
+		if resp.DrainProgressDescription != "" {
+			// Only show this information in the log; we'd use this for debugging.
+			// (This can be revealed e.g. via --logtostderr.)
+			log.Infof(ctx, "drain details: %s\n", resp.DrainProgressDescription)
+		}
+
+		if resp.IsDraining {
+			// To report the status of the drain, we have two cases:
+			//
+			// - in the initial drain request sent by quit (with shutdown =
+			//   false) we want to assert that the node is quitting, and
+			//   tell the story about how much work was performed in logs
+			//   for debugging. This is also the only thing that happens
+			//   with --only-drain.
+			//
+			// - in the final drain request sent by quit (with shutdown = true)
+			//   we want to assert that the progress indicator is zero,
+			//   otherwise something may be afoul in the server.
+			//
+			// In any case we want to print on stderr or in logs, not
+			// stdout. That's because stdout's interface should just be "ok"
+			// on success (otherwise redirects could confuse scripts).
+			log.Infof(ctx, "drain progress indicator: %d\n", resp.DrainProgressIndicator)
+			if !requestNodeShutdown {
+				// initial request. If --only-drain was specified,
+				// tell the user how much work was performed.
+				// The user can then re-run quit --only-drain multiple
+				// times to see the value decrease.
+				if quitCtx.onlyDrain {
+					finalString := ""
+					if resp.DrainProgressIndicator == 0 {
+						finalString = " (complete)"
+					}
+					fmt.Fprintf(stderr, "# node is shutting down; progress: %d%s\n",
```

Comment:
Nice, I'm really happy to see progress reporting here. However, I think it might be misleading to say "progress: 12" here. We tend to think of progress as something that goes up (perhaps finishing once it reaches 100, or something like that). Consider this instead:

> node is shutting down; remaining: %d

which makes clear that whatever the %d is, we're done when it reaches zero.

Reply:
(similarly DrainProgressIndicator is a misnomer)